Core/Src/main.c

```diff
@@ -55,7 +57,7 @@ void SystemClock_Config(void);
 
 /* Private user code ---------------------------------------------------------*/
 /* USER CODE BEGIN 0 */
-
+uint16_t read_adc();
```

Comment:
great use of a function, makes things very organized in main. One thing to note here, in C, although read_adc() and read_adc(void) would yield the exactly same output, they are TECHNICALLY very different. When you add `void` you are explicitly saying this function takes NO parameters, but the same cannot be said about `read_adc()`, in this case, you are specifying that the function can take ANY number (including no) parameters. So the more readable option would be read_adc(void). This is not the case in C++ as it actually is the same.

If you would like to test, call your read_adc() function with random parameters, it should still work.